### PR TITLE
Feature/interface for material upload

### PR DESCRIPTION
**Pull request by Adem-Khachnaoui**

### CODE REVIEW

**Comment by CanArisan** [comment]
Hey @Adem-Khachnaoui , I managed to fix the error I got while testing this. I was getting a "Certificate verify failed" error while executing nltk.download(). I found this in the internet and added it to data_cleaner.py in my local branch:
Screenshot 2020-06-08 at 12 03 11

Now it works fine by me, nice job! I don't know if we should add this to the code though. If you want I can push this, or leave it as it is, as it works with you.

**Comment by Adem-Khachnaoui** [comment]
> Hey @Adem-Khachnaoui , I managed to fix the error I got while testing this. I was getting a "Certificate verify failed" error while executing nltk.download(). I found this in the internet and added it to data_cleaner.py in my local branch:
> Screenshot 2020-06-08 at 12 03 11
> 
> Now it works fine by me, nice job! I don't know if we should add this to the code though. If you want I can push this, or leave it as it is, as it works with you.
@CanArisan, great! if it solved your problem maybe it's better that you push it because others  might have the same issue.

**Comment by CanArisan** [comment]
> Hey @Adem-Khachnaoui , I managed to fix the error I got while testing this. I was getting a "Certificate verify failed" error while executing nltk.download(). I found this in the internet and added it to data_cleaner.py in my local branch:
> Screenshot 2020-06-08 at 12 03 11
> 
> Now it works fine by me, nice job! I don't know if we should add this to the code though. If you want I can push this, or leave it as it is, as it works with you.

Oh sorry, I just realised this is the wrong branch. I meant this for your benchmark branch https://github.com/ls1intum/Athene/pull/10

**Comment by Adem-Khachnaoui** [comment]
> I tested along with the PR in Artemis, it works good +1
> And maybe we should add some sort of error handling for cloud related operations. What do you think?

I added more error handling (wrong credentials and error in file uplaod)

**Comment by bigultekin** [comment]
> > I tested along with the PR in Artemis, it works good +1
> > And maybe we should add some sort of error handling for cloud related operations. What do you think?
> 
> Thanks for the feedback!
> I added more error handling (wrong credentials and error in file upload)

Nice, it looks better 👍

File: embedding/Makefile [hunk 1]
```diff
@@ -1,2 +1,2 @@
 deps:
-	@$(MAKE) -C src/resources;
\ No newline at end of file
+	@$(MAKE) -C src/resources/models;
```
Comment (linusmichel) [inline embedding/Makefile:2]: I think we should add newlines at the end of all files for compatibility

File: embedding/src/cloud/config.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+nextcloud_credentials = {
+    "login": "",
+    "password": ""
+}
```
Comment (linusmichel) [inline embedding/src/cloud/config.py:4]: I think we should add newlines at the end of all files for compatibility

File: embedding/src/resources/models/Makefile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# https://allennlp.org/elmo
+
+deps: elmo_2x4096_512_2048cnn_2xhighway_5.5B
+elmo_2x4096_512_2048cnn_2xhighway_5.5B: elmo_2x4096_512_2048cnn_2xhighway_5.5B_weights.hdf5 elmo_2x4096_512_2048cnn_2xhighway_5.5B_options.json
+
+elmo_2x4096_512_2048cnn_2xhighway_5.5B_weights.hdf5:
+	@curl -OJ "https://nextcloud.in.tum.de/index.php/s/xB7QitYX2ydbSpC/download"
+
+elmo_2x4096_512_2048cnn_2xhighway_5.5B_options.json:
+	@curl -OJ "https://nextcloud.in.tum.de/index.php/s/itNpqDHE8NqpBgy/download"
```
Comment (linusmichel) [inline embedding/src/resources/models/Makefile:11]: I think we should add newlines at the end of all files for compatibility

File: embedding/src/errors.py [hunk 4]
```diff
@@ -1,4 +1,7 @@
 from falcon import HTTPBadRequest
 
 emptyBody = HTTPBadRequest("Empty Requerst Body", "Cannot process request with empty body.")
-requireTwoBlocks = HTTPBadRequest("Need two or more blocks", "Must provide at least two text blocks.")
\ No newline at end of file
+requireTwoBlocks = HTTPBadRequest("Need two or more blocks", "Must provide at least two text blocks.")
+requireCourseId = HTTPBadRequest("CourseId not provided", "Must provide a course identifier to upload material")
+requireFileData = HTTPBadRequest("File data is not provided", "Must provide file data to upload material")
+requireFileName = HTTPBadRequest("File name is not provided", "Must provide file name to upload material")
```
Comment (linusmichel) [inline embedding/src/errors.py:7]: I think we should add newlines at the end of all files for compatibility

File: embedding/src/cloud/cloud_connection.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
+from logging import getLogger
+
+from owncloud import owncloud, HTTPResponseError
+
+from .config import nextcloud_credentials
```
Comment (linusmichel) [inline embedding/src/cloud/cloud_connection.py:5]: I would not leave newlines between imports

File: embedding/src/UploadingRessource.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import base64
+import json
+from datetime import datetime
+from logging import getLogger
+from typing import List
+
+from falcon import Request, Response, HTTP_200
+
+from src.cloud.cloud_connection import CloudConnection
+from src.errors import emptyBody, requireCourseId, requireFileName, requireFileData
```
Comment (linusmichel) [inline embedding/src/UploadingRessource.py:10]: I would not leave newlines between imports

File: embedding/README.md [hunk 7]
```diff
@@ -21,3 +26,15 @@ Input example JSON for POST
   ]
  }
 ```
+
+
+Input example JSON for POST on http://localhost:8001/upload
```
Comment (linusmichel) [inline embedding/README.md:31]: Is it possible to use the API with other files than PDF's like docx,pptx,txt, etc.?

Comment (Adem-Khachnaoui) [reply embedding/README.md:31]: yes, it is possible, But Artemis will only send PDFs for the moment

**Comment by Adem-Khachnaoui** [comment]
> Looks very good to me. Just commented on some formal things.

Changes applied! thanks for the review

**Pull request merged**